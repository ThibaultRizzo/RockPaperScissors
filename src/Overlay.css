#Overlay {
    display: flex;
    flex-flow: row nowrap;
    justify-content: space-around;
    align-items: center;
}

.overlay-button.special-button {
    flex-basis: 30%;
    height: 100px;
}

.Overlay {
    height: 100%;
    width: 0;
    position: fixed;
    z-index: 1;
    top: 0;
    left: 0;
    background-color: rgb(0,0,0);
    background-color: rgba(0,0,0, 0.9);
    overflow-x: hidden;
    transition: 0.5s;
}
