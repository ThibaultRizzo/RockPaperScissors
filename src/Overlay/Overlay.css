#Overlay {
  display: flex;
  flex-flow: row wrap;
  justify-content: space-around;
  align-items: center;
}

.overlay-button.special-button {
  flex-basis: 30%;
  height: 100px;
}

.turn-dropdown {
  display: flex;
  flex-flow: column nowrap;
  justify-content: flex-start;
  align-items: stretch;
  flex-basis: 10%;
  margin: 0 30%;
  width: 15%;
  height: 20%;
}

input.selected-dropdown-value {
  flex: 0 0 30%;
  font-size: 30px;
}

.dropdown-value {
  background-color: white;
  border-radius: 25%;
  box-shadow: black 0px 2px 2px 2px;
  border: 2px solid black;
  margin-bottom: 8px;
  text-align: center;
}

.turn-option {
  flex-basis: 20%;
}
