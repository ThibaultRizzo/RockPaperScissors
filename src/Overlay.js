import React, { Component } from 'react';
import './Overlay.css';

class TurnDropDown extends Component {
  constructor(props) {
    super(props);
    this.toggleDropdown = this.toggleDropdown.bind(this);
    this.onSelectTurn = this.onSelectTurn.bind(this);
    this.state = {
      currentTurn: this.props.currentTurn,
      isDropdownExpanded: false
    };
  }
  onSelectTurn = e => {
    debugger;
    this.setState({
      isDropdownExpanded: false
    });
    this.props.selectTurn(parseInt(e, 10));
  };

  toggleDropdown = e => {
    this.setState({
      isDropdownExpanded: !this.state.isDropdownExpanded,
      currentTurn: this.props.currentTurn
    });
  };
  render() {
    let turnChoices;
    if (this.state.isDropdownExpanded) {
      turnChoices = [];
      for (let i = 3; i <= this.props.maxTurn; i++) {
        turnChoices.push(
          <div
            key={i}
            className="turn-option dropdown-value"
            name={i}
            value={i}
            onClick={this.onSelectTurn}
          >
            {i}
          </div>
        );
      }
    } else {
      turnChoices = false;
    }

    return (
      <div className="turn-dropdown">
        <input
          type="button"
          className="selected-dropdown-value dropdown-value"
          value={this.props.currentTurn}
          onClick={this.toggleDropdown}
        />
        <div className="dropdown-content">{turnChoices}</div>
      </div>
    );
  }
}

class Overlay extends Component {
  MAX_TURN = 5;
  DEFAULT_TURN = 3;
  constructor(props) {
    super(props);
    this.state = {
      currentTurn: this.DEFAULT_TURN
    };
  }
  setGameType = (currentTurn, e) => {
    this.props.onOptionChosed(currentTurn, e.target.value);
  };

  selectTurn = e => {
    this.setState({
      currentTurn: e.target.textContent,
      isDropdownExpanded: false
    });
  };
  render() {
    return (
      <div id="Overlay" className="full-page-background smooth-appear">
        <input
          type="button"
          value="Player vs AI"
          className="overlay-button special-button"
          onClick={e => this.setGameType(this.state.currentTurn, e)}
        />
        <input
          type="button"
          value="AI vs AI"
          className="overlay-button special-button"
          onClick={e => this.setGameType(this.state.currentTurn, e)}
        />
        <TurnDropDown
          selectTurn={this.selectTurn}
          maxTurn={this.MAX_TURN}
          currentTurn={this.state.currentTurn}
        />
      </div>
    );
  }
}

export default Overlay;
