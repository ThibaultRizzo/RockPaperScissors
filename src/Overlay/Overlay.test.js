import React from 'react';
import ReactDOM from 'react-dom';
import Overlay from './Overlay';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const clicked = jest.fn();
  const rendered = renderer.create(<Overlay onOptionChosed={clicked} />);
  expect(rendered.toJSON()).toMatchSnapshot();
});
