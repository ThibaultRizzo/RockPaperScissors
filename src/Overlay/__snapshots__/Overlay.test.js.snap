// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="full-page-background smooth-appear"
  id="Overlay"
>
  <input
    className="overlay-button special-button"
    onClick={[Function]}
    type="button"
    value="Player vs AI"
  />
  <input
    className="overlay-button special-button"
    onClick={[Function]}
    type="button"
    value="AI vs AI"
  />
  <div
    className="turn-dropdown"
  >
    <input
      className="selected-dropdown-value dropdown-value"
      onClick={[Function]}
      type="button"
      value={3}
    />
    <div
      className="dropdown-content"
    />
  </div>
</div>
`;
